### PR TITLE
Add generics slides

### PR DESCRIPTION
**Pull request by jonathanpallant**
Add some slides on generics. I tried to make it not too complicated.

### CODE REVIEW

File: training-slides/src/generics.md
Comment (amanjeev) [inline training-slides/src/generics.md:1]: it seems like you have added changes to this PR from the methods and traits slides ? https://github.com/ferrous-systems/rust-training/pull/62

File: training-slides/src/generics.md [hunk 1]
```diff
@@ -1 +1,290 @@
...
+        self.area_m2 += shape.area_m2();
+    }
+}
+```
+
+## Caution
+
+* Using Generics is *Hard Mode Rust*
+* Don't reach for it in the first instance...
+  * Try and just use concrete types?
```
Comment (amanjeev) [inline training-slides/src/generics.md:269]:
```suggestion
  * Try and just use concrete types
```

Not sure if that ? is needed. Ignore if it is.

Comment (jonathanpallant) [reply training-slides/src/generics.md:269]: It's to help you read it with an upward Australian style "I'm not quite sure about this"  inflection?

File: training-slides/src/generics.md
Comment (jonathanpallant) [reply training-slides/src/generics.md:1]: Yes. I did some work, cut a branch and made a PR, then carried on working. Sorry about that.

**Pull request merged**